src/measure_theory/constructions/polish.lean

```diff
@@ -0,0 +1,689 @@
+    have cauchy_y : cauchy_seq y,
+    { have : tendsto (λ n, 2 * u n) at_top (𝓝 0), by simpa only [mul_zero] using u_lim.const_mul 2,
+      apply cauchy_seq_of_le_tendsto_0' (λ n, 2 * u n) (λ m n hmn, _) this,
+      rcases I m n with ⟨z, zsm, zsn⟩,
+      calc dist (y m) (y n) ≤ dist (y m) z + dist z (y n) : dist_triangle _ _ _
+      ... ≤ u m + u n :
+        add_le_add ((dist_le_diam_of_mem (hs m).1 (hy m) zsm).trans (hs m).2)
+                   ((dist_le_diam_of_mem (hs n).1 zsn (hy n)).trans (hs n).2)
+      ... ≤ 2 * u m : by linarith [u_anti.antitone hmn] },
+    haveI : nonempty α := ⟨y 0⟩,
+    -- let `z` be its limit.
+    let z := lim at_top y,
+    have y_lim : tendsto y at_top (𝓝 z) := cauchy_y.tendsto_lim,
+    suffices : f z = x, by { rw ← this, exact mem_range_self _ },
+    -- assume for a contradiction that `f z ≠ x`.
+    by_contra' hne,
+    -- introduce disjoint open sets `v` and `w` separating `f z` from `x`.
+    obtain ⟨v, w, v_open, w_open, fzv, xw, hvw⟩ :
+      ∃ v w : set β, is_open v ∧ is_open w ∧ f z ∈ v ∧ x ∈ w ∧ v ∩ w = ∅ :=
+        t2_separation hne,
+    obtain ⟨δ, δpos, hδ⟩ : ∃ δ > (0 : ℝ), ball z δ ⊆ f ⁻¹' v,
+    { apply metric.mem_nhds_iff.1,
+      exact f_cont.continuous_at.preimage_mem_nhds (v_open.mem_nhds fzv) },
+    obtain ⟨n, hn⟩ : ∃ n, u n + dist (y n) z < δ,
+    { have : tendsto (λ n, u n + dist (y n) z) at_top (𝓝 0),
+        by simpa only [add_zero] using u_lim.add (tendsto_iff_dist_tendsto_zero.1 y_lim),
+      exact ((tendsto_order.1 this).2 _ δpos).exists },
+    -- for large enough `n`, the image of `s n` is contained in `v`, by continuity of `f`.
+    have fsnv : f '' (s n) ⊆ v,
+    { rw image_subset_iff,
+      apply subset.trans _ hδ,
+      assume a ha,
+      calc dist a z ≤ dist a (y n) + dist (y n) z : dist_triangle _ _ _
+      ... ≤ u n + dist (y n) z :
+        add_le_add_right ((dist_le_diam_of_mem (hs n).1 ha (hy n)).trans (hs n).2) _
+      ... < δ : hn },
+    -- as `x` belongs to the closure of `f '' (s n)`, it belongs to the closure of `v`.
+    have : x ∈ closure v := closure_mono fsnv (hxs n).1,
+    -- this is a contradiction, as `x` is supposed to belong to `w`, which is disjoint from
+    -- the closure of `v`.
+    exact disjoint_left.1 ((disjoint_iff_inter_eq_empty.2 hvw).closure_left w_open) this xw }
+end
+
+theorem _root_.is_closed.measurable_set_image_of_continuous_on_inj_on [polish_space α]
+  {β : Type*} [topological_space β] [t2_space β] [measurable_space β] [borel_space β]
+  {s : set α} (hs : is_closed s) {f : α → β} (f_cont : continuous_on f s) (f_inj : inj_on f s) :
+  measurable_set (f '' s) :=
+begin
+  rw image_eq_range,
+  haveI : polish_space s := is_closed.polish_space hs,
+  apply measurable_set_range_of_continuous_injective,
+  { rwa continuous_on_iff_continuous_restrict at f_cont },
+  { rwa inj_on_iff_injective at f_inj }
+end
+
+/-- The Lusin-Souslin theorem: if `s` is Borel-measurable in a Polish space, then its image under
```

Comment:
Can you add a `variables` declaration here (possibly in a section) to state that alpha and beta are polish borel spaces?